### PR TITLE
Make sure angleconstraints only take angles between 0 and 180 degrees.

### PR DESCRIPTION
**Pull request by nmaeder**
#### What does this implement/fix? Explain your changes.
Make sure AngleConstraints can only take min and max angles between 0.0 and 180.
Using negative angles or angles > 180.0 (i.e. angle - 360) leads to the angles minimized to 0.0
Range checks added to prevent this from happening, and removing tests.

Since both the MMFF and UFF angle constraint are the same, the `AngleConstraint.h` and `AngleConstraint.cpp` are moved to the ForceField. To ensure backwards compatibility, the headers remain in their respective folders. 

The way the angles are calculated in `AngleConstraint.cpp` are unified. 

#### Any other comments?
Here is code to reproduce the weird behavior with the current master:

```python
from rdkit import Chem
from rdkit.Chem import ChemicalForceFields, rdDistGeom, rdMolTransforms

water = Chem.AddHs(Chem.MolFromSmiles("O"))
rdDistGeom.EmbedMolecule(water)
ff = ChemicalForceFields.CreateEmptyForceFieldForMol(water)
ff.MMFFAddAngleConstraint(1,0,2,False, 190,190,1)
# ff.MMFFAddAngleConstraint(1,0,2,False, -90,-90,1) 
ff.Minimize()
```

### CODE REVIEW

File: Code/ForceField/AngleConstraint.cpp [hunk 1]
```diff
@@ -52,15 +49,14 @@ AngleConstraintContrib::AngleConstraintContrib(
...
-    RDGeom::Point3D p3 = *((RDGeom::Point3D *)pos[idx3]);
-    double dist1 = (p1 - p2).length();
-    double dist2 = (p3 - p2).length();
-    RDGeom::Point3D p12 = (p1 - p2) / dist1;
-    RDGeom::Point3D p32 = (p3 - p2) / dist2;
-    double cosTheta = p12.dotProduct(p32);
-    clipToOne(cosTheta);
+    const RDGeom::Point3D p1 = *((RDGeom::Point3D *)pos[idx1]);
+    const RDGeom::Point3D p2 = *((RDGeom::Point3D *)pos[idx2]);
+    const RDGeom::Point3D p3 = *((RDGeom::Point3D *)pos[idx3]);
```
Comment (greglandrum) [inline Code/ForceField/AngleConstraint.cpp:54]:
```suggestion
    const RDGeom::Point3D &p1 = *((RDGeom::Point3D *)pos[idx1]);
    const RDGeom::Point3D &p2 = *((RDGeom::Point3D *)pos[idx2]);
    const RDGeom::Point3D &p3 = *((RDGeom::Point3D *)pos[idx3]);
```

File: Code/ForceField/AngleConstraint.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include <RDGeneral/export.h>
+#ifndef __RD_ANGLECONSTRAINT_H__
+#define __RD_ANGLECONSTRAINT_H__
```
Comment (greglandrum) [inline Code/ForceField/AngleConstraint.h:12]: We used to prefix these with `__`, but it's considered bad style, so I change them whenever I notice
```suggestion
#ifndef RD_ANGLECONSTRAINT_H
#define RD_ANGLECONSTRAINT_H
```

File: Code/ForceField/Params.h
Comment (greglandrum) [inline Code/ForceField/Params.h:1]: Why is this here?
I'm not sure we need it all and nothing in here is a parameter.

File: Code/ForceField/MMFF/testMMFFForceField.cpp [hunk 3]
```diff
@@ -1513,20 +1513,6 @@ void testMMFFAllConstraints() {
       MolTransforms::getAngleDeg(mol->getConformer(), 1, 3, 6), 100.0, 0.5));
   delete field;
   delete mmffMolProperties;
-  MolTransforms::setAngleDeg(mol->getConformer(), 1, 3, 6, 0.0);
```
Comment (greglandrum) [inline Code/ForceField/MMFF/testMMFFForceField.cpp:1516]: I would keep this test, but use the new AngleConstraintContrib

File: Code/ForceField/UFF/testUFFForceField.cpp [hunk 4]
```diff
@@ -1414,16 +1414,6 @@ void testUFFAllConstraints() {
   TEST_ASSERT(RDKit::feq(
       MolTransforms::getAngleDeg(mol->getConformer(), 1, 3, 6), 100.0, 0.5));
   delete field;
-  MolTransforms::setAngleDeg(mol->getConformer(), 1, 3, 6, 0.0);
```
Comment (greglandrum) [inline Code/ForceField/UFF/testUFFForceField.cpp:1417]: same comment as above: please keep the test, but use the new AngleConstraintContrib

File: Code/ForceField/Wrap/testConstraints.py [hunk 5]
```diff
@@ -157,14 +157,6 @@ def testUFFAngleConstraints(self):
     conf = m.GetConformer()
     angle = rdMolTransforms.GetAngleDeg(conf, 1, 3, 6)
     self.assertAlmostEqual(angle, 100, delta=0.1)
-    ff = ChemicalForceFields.UFFGetMoleculeForceField(m)
```
Comment (greglandrum) [inline Code/ForceField/Wrap/testConstraints.py:160]: The tests in this file probably shouldn't be changed

**Review by greglandrum — CHANGES_REQUESTED** [review]
a number of changes suggested

File: Code/ForceField/Params.h
Comment (nmaeder) [reply Code/ForceField/Params.h:1]: moved the content to `AngleConstraint.cpp`.

**Comment by nmaeder** [comment]
requested changes incorporated

File: Code/ForceField/AngleConstraint.cpp [hunk 6]
```diff
@@ -69,8 +68,8 @@ AngleConstraintContrib::AngleConstraintContrib(
   d_at3Idx = idx3;
   minAngleDeg += angle;
   maxAngleDeg += angle;
-  RDKit::ForceFieldsHelper::normalizeAngleDeg(minAngleDeg);
-  RDKit::ForceFieldsHelper::normalizeAngleDeg(maxAngleDeg);
+  RANGE_CHECK(0.0, minAngleDeg, 180.0);
+  RANGE_CHECK(0.0, maxAngleDeg, 180.0);
```
Comment (greglandrum) [inline Code/ForceField/AngleConstraint.cpp:72]: I think these lines can move inside the `if(relative)` loop

File: Code/ForceField/AngleConstraint.cpp [hunk 7]
```diff
@@ -52,15 +52,14 @@ AngleConstraintContrib::AngleConstraintContrib(
 
   double angle = 0.0;
```
Comment (greglandrum) [inline Code/ForceField/AngleConstraint.cpp:53]: This can move inside the `if(relative)` loop if you make the other suggested change below

**Review by greglandrum — CHANGES_REQUESTED** [review]
Last set of suggested changes, I promise. :-)

File: Code/ForceField/AngleConstraint.cpp (hunk 6 above)
Comment (nmaeder) [reply Code/ForceField/AngleConstraint.cpp:72]: line 69-70 definitely, the `RANGE_CHECK` however should stay outside, otherwise if `relative` is `false` we could still set negative values.

File: Code/ForceField/AngleConstraint.cpp (hunk 7 above)
Comment (nmaeder) [reply Code/ForceField/AngleConstraint.cpp:53]: yes, if the `+=` go inside, then this is unnecessary and we can just define it when we calculate it.

File: Code/ForceField/AngleConstraint.cpp (hunk 6 above)
Comment (greglandrum) [reply Code/ForceField/AngleConstraint.cpp]: you can modify minAngleDeg and maxAngleDeg inside the loop too

**Review by greglandrum — APPROVED** [review]
LGTM
